contracts/governance/Governor.sol

```diff
@@ -207,14 +203,16 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor, IERC721Receive
      * @dev See {IGovernor-proposalSnapshot}.
      */
     function proposalSnapshot(uint256 proposalId) public view virtual override returns (uint256) {
-        return _proposals[proposalId].voteStart;
+        ProposalCore memory proposal = _proposals[proposalId];
+        return proposal.voteStart;
     }
 
     /**
      * @dev See {IGovernor-proposalDeadline}.
      */
     function proposalDeadline(uint256 proposalId) public view virtual override returns (uint256) {
-        return _proposals[proposalId].voteEnd;
+        ProposalCore memory proposal = _proposals[proposalId];
+        return proposal.voteStart + proposal.voteDuration;
```

Comment:
This makes me wonder if Solidity would optimize the following:

```
ProposalCore storage proposal = _proposals[proposalId];
return proposal.voteStart + proposal.voteDuration;
```

into a single SLOAD.... That would be better because it saves an allocation.

Reply:
I don't trust the compiler to go that, but yes we need to check.

That might be one of these case where the "optimal" solidity code depends on the compiler settings.

Reply:
```suggestion
        return _proposals[proposalId].voteStart + _proposals[proposalId].voteDuration;
```